Sources/MeowVapor/MeowProvider.swift

```diff
@@ -12,20 +12,48 @@ public final class MeowProvider: Provider {
         return .done(on: container)
     }
     
+    let lazy: Bool
     let connectionSettings: ConnectionSettings
     
-    public init(_ uri: String) throws {
+    /// Connects to the MongoDB server or cluster located at the URI
+    ///
+    /// If `lazy` is set to true, the first error Meow throws will occur on the first query, not when creating the database, manager or context.
+    /// The advantage of using `lazy` is the ability to call `request.make(Meow.Context.self)` and `request.make(Meow.Manager.self)`
+    ///
+    /// For backwards compatibility and predictability, lazy defaults to `false`
+    public init(_ uri: String, lazy: Bool = false) throws {
```

Comment:
I think that it is more sensible to make `lazy` default to true.

For backwards compatibility, I would then add a second (convenience) `init` with the current signature that is deprecated.

I would then change this init to `init(uri: String, lazy: Bool = true)`. I think the `uri` argument label is better for clarity anyway, so that the syntax becomes `MeowProvider(uri: "mongodb://...")`